src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+
+                    var newComment = new ClientComment
+                    {
+                        ClientId = clientId,
+                        Comment = commentText,
+                        CreatedAt = DateTime.UtcNow,
+                        FullName = "Lykke.Job.TransactionHandler",
+                        UserId = transactionHandlerUserId
+                    };
+
+                    var exResult = await _exchangeOperationsServiceClient.ManualCashInAsync(clientId, assetId,
+                            (double)amount, "auto redeem", commentText);
+
+                    if (!exResult.IsOk())
+                    {
+                        await
+                            _log.WriteWarningAsync(nameof(EthereumCoreCommandHandler), nameof(ProcessFailedCashout),
+                            (new
+                            {
+                                ExchangeServiceResponse = exResult,
+                                QueueMessage = queueMessage
+                            }).ToJson(),
+                                "ME operation failed");
+
+                        return false;
+                    }
+
+                    await _clientCommentsRepository.AddClientCommentAsync(newComment);
+                    await _paymentTransactionsRepository.SetStatus(hash, PaymentStatus.NotifyProcessed);
+                }
+                catch (Exception e)
+                {
+                    await _log.WriteErrorAsync(nameof(EthereumCoreCommandHandler), nameof(ProcessFailedCashout), queueMessage.ToJson(), e);
+                    throw e;
+                }
+            }
+            else
+            {
+                await _log.WriteWarningAsync(nameof(EthereumCoreCommandHandler), nameof(ProcessFailedCashout), queueMessage.ToJson());
+            }
+
+            return true;
+        }
+
+        private async Task SetTradeHashes(IEthereumTransactionRequest txRequest, string hash)
+        {
+            foreach (var id in txRequest.OperationIds)
+            {
+                await _clientTradesRepositoryClient.UpdateBlockchainHashAsync(txRequest.ClientId, id, hash);
+            }
+        }
+
+        private async Task SetCashoutHashes(IEthereumTransactionRequest txRequest, string hash)
+        {
+            foreach (var id in txRequest.OperationIds)
+            {
+                await _cashOperationsRepositoryClient.UpdateBlockchainHashAsync(txRequest.ClientId, id, hash);
+            }
+        }
+
+        private async Task SetTransferHashes(IEthereumTransactionRequest txRequest, string hash)
+        {
+            foreach (var id in txRequest.OperationIds)
+            {
+                await _transferEventsRepositoryClient.UpdateBlockChainHashAsync(txRequest.ClientId, id, hash);
+            }
+        }
+
+        private async Task<bool> ProcessCashIn(ProcessCoinEventCommand queueMessage, IEventPublisher eventPublisher)
```

Comment:
Result is always true and it's not used by the caller - should be void

Reply:
fixed